services/src/main/java/org/keycloak/social/github/GitHubIdentityProvider.java

```diff
@@ -35,18 +35,79 @@
  */
 public class GitHubIdentityProvider extends AbstractOAuth2IdentityProvider implements SocialIdentityProvider {
 
-	public static final String AUTH_URL = "https://github.com/login/oauth/authorize";
-	public static final String TOKEN_URL = "https://github.com/login/oauth/access_token";
-	public static final String PROFILE_URL = "https://api.github.com/user";
-	public static final String EMAIL_URL = "https://api.github.com/user/emails";
-	public static final String DEFAULT_SCOPE = "user:email";
-
-	public GitHubIdentityProvider(KeycloakSession session, OAuth2IdentityProviderConfig config) {
-		super(session, config);
-		config.setAuthorizationUrl(AUTH_URL);
-		config.setTokenUrl(TOKEN_URL);
-		config.setUserInfoUrl(PROFILE_URL);
-	}
+    public static final String DEFAULT_BASE_URL = "https://github.com";
+    public static final String AUTH_FRAGMENT = "/login/oauth/authorize";
+    public static final String TOKEN_FRAGMENT = "/login/oauth/access_token";
+    public static final String DEFAULT_AUTH_URL = DEFAULT_BASE_URL + AUTH_FRAGMENT;
+    public static final String DEFAULT_TOKEN_URL = DEFAULT_BASE_URL + TOKEN_FRAGMENT;
+    /** @deprecated Use {@link #DEFAULT_AUTH_URL} instead. */
+    @Deprecated
+    public static final String AUTH_URL = DEFAULT_AUTH_URL;
+    /** @deprecated Use {@link #DEFAULT_TOKEN_URL} instead. */
+    @Deprecated
+    public static final String TOKEN_URL = DEFAULT_TOKEN_URL;
+
+    public static final String DEFAULT_API_URL = "https://api.github.com";
+    public static final String PROFILE_FRAGMENT = "/user";
+    public static final String EMAIL_FRAGMENT = "/user/emails";
+    public static final String DEFAULT_PROFILE_URL = DEFAULT_API_URL + PROFILE_FRAGMENT;
+    public static final String DEFAULT_EMAIL_URL = DEFAULT_API_URL + EMAIL_FRAGMENT;
+    /** @deprecated Use {@link #DEFAULT_PROFILE_URL} instead. */
+    @Deprecated
+    public static final String PROFILE_URL = DEFAULT_PROFILE_URL;
+    /** @deprecated Use {@link #DEFAULT_EMAIL_URL} instead. */
+    @Deprecated
+    public static final String EMAIL_URL = DEFAULT_EMAIL_URL;
```

Comment:
that is right, EMAIL_URL is not used in this class, but since it is public, I didn't want to remove it, hence I deprecated in flavor is better named DEFAULT_EMAIL_URL